src/main/java/uk/gov/hmcts/ccd/domain/service/message/CaseEventMessageService.java

```diff
@@ -0,0 +1,84 @@
+import uk.gov.hmcts.ccd.domain.model.std.MessageQueueCandidate;
+
+import javax.inject.Inject;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.List;
+
+@Service
+@Qualifier("default")
+public class CaseEventMessageService implements MessageService {
+
+    private static final ObjectMapper mapper = new ObjectMapper();
+    private final UserRepository userRepository;
+    private final MessageCandidateRepository messageCandidateRepository;
+    private final CaseAuditEventRepository caseAuditEventRepository;
+    private static String CASE_EVENT_MESSAGE_TYPE = "CASE_EVENT";
+
+    @Inject
+    public CaseEventMessageService(@Qualifier(CachedUserRepository.QUALIFIER) final UserRepository userRepository,
+                                   final MessageCandidateRepository messageCandidateRepository,
+                                   CaseAuditEventRepository caseAuditEventRepository) {
+        this.userRepository = userRepository;
+        this.messageCandidateRepository = messageCandidateRepository;
+        this.caseAuditEventRepository = caseAuditEventRepository;
+    }
+
+    @Override
+    public void handleMessage(Event event,
+                              CaseEventDefinition caseEventDefinition,
+                              CaseDetails caseDetails) {
+        final MessageQueueCandidate messageQueueCandidate = new MessageQueueCandidate();
+        if (Boolean.TRUE.equals(caseEventDefinition.getPublish())) {
+
+            MessageInformation messageInformation = populateMessageInformation(event, caseEventDefinition, caseDetails);
+            JsonNode node = mapper.convertValue(messageInformation, JsonNode.class);
+
+            messageQueueCandidate.setMessageInformation(node);
+            messageQueueCandidate.setMessageType(CASE_EVENT_MESSAGE_TYPE);
+            messageQueueCandidate.setTimeStamp(LocalDateTime.now(ZoneOffset.UTC));
+            messageCandidateRepository.set(messageQueueCandidate);
+        }
+    }
+
+    private MessageInformation populateMessageInformation(Event event,
+                                                          CaseEventDefinition caseEventDefinition,
+                                                          CaseDetails caseDetails) {
+
+        final MessageInformation messageInformation = new MessageInformation();
+        final IdamUser user = userRepository.getUser();
+        List<AuditEvent> auditEvent = caseAuditEventRepository.findByCase(caseDetails);
+
+        messageInformation.setCaseId(caseDetails.getReference().toString());
+        messageInformation.setJurisdictionId(caseDetails.getJurisdiction());
+        messageInformation.setCaseTypeId(caseDetails.getCaseTypeId());
+        messageInformation.setEventInstanceId(auditEvent.get(0).getId().toString());
+        messageInformation.setEventTimestamp(caseDetails.getLastStateModifiedDate());
+        messageInformation.setEventId(event.getEventId());
```

Comment:
Can `EventId` be obtained from `CaseEventDefinition`, then we can remove `Event` from the method signature completely?

Reply:
done 👍